addons/website_event/static/src/js/components/navbar/navbar.js

```diff
@@ -1,23 +1,21 @@
 /** @odoo-module  */
 
-import { patch } from 'web.utils';
-import { NavBar } from '@web/webclient/navbar/navbar';
+import { patch } from '@web/core/utils/patch';
+import { websiteCustomMenus } from '@website/components/navbar/navbar';
 import { EditMenuDialog } from '@website/components/dialog/edit_menu';
 
-const { onWillStart } = owl;
-
-patch(NavBar.prototype, 'website_events_navbar', {
-    setup() {
-        this._super();
-
-        onWillStart(() => {
-            this.websiteEditingMenus['website_event.menu_edit_menu'] = {
-                Component: EditMenuDialog,
-                isDisplayed: () => this.websiteService.currentWebsite && this.websiteService.currentWebsite.metadata.contentMenuId,
-                getProps: () => ({
-                    rootID: parseInt(this.websiteService.currentWebsite.metadata.contentMenuId, 10),
-                }),
-            };
-        });
+patch(websiteCustomMenus, 'website_events_custom_menus', {
+    start(env, { website, ui }) {
+        const service = this._super(...arguments);
+        service.menus['website_event.menu_edit_menu'] = {
+            Component: EditMenuDialog,
+            isDisplayed: () => website.currentWebsite
+                && website.currentWebsite.metadata.contentMenuId
+                && !ui.isSmall,
+            getProps: () => ({
+                rootID: parseInt(website.currentWebsite.metadata.contentMenuId, 10),
+            }),
+        };
+        return service;
```

Comment:
Do you actually need to patch your service like that? Is not there a prettier way to add entries to a service?

Reply:
I pushed a new version of the commit https://github.com/odoo/odoo/commit/182b63cc0f1592b1b8e1cded609c5754a8e535a3, so that menus can be added using a registry